### PR TITLE
Tgcc executor

### PR DESCRIPTION
**Pull request by lindenb**
I currently work with the TGCC supercomputing center https://www-hpc.cea.fr/en/TGCC.html

> TGCC is an infrastructure for scientific high-performance computing and Big Data, able to host petascale supercomputers. This supercomputing center has been planned to welcome the first French Petascale machine Curie, funded by GENCI for the PRACE Research Infrastructure, and the next generation of Computing Center for Research and Technology (CCRT).

they use a scheduler `ccc` built over `slurm` (eg: `ccc_msub` instead of `sbatch`).  Among other things, each submission require the user to specify a project. 

So, this potential PR, implements the TgccExecutor extending the SlurmExecutor . If it's ok with you I'll update the doc, fix the indentation , etc...

For now it looks like it's working... 
```
Jul-31 16:50:31.400 [Task submitter] DEBUG nextflow.executor.GridTaskHandler - [TGCC] submitted process MAP:BWA ....
```

The TGCC is isolated from the web so I can only use the packed executable.

I'm not sure if a test is needed ?

I'm waiting for your go/no-go :slightly_smiling_face: 

best regards,

Pierre

### CODE REVIEW

File: modules/nextflow/src/main/groovy/nextflow/executor/TgccExecutor.groovy [hunk 1]
```diff
@@ -0,0 +1,200 @@
+/*
+ * Copyright 2013-2023, Seqera Labs
```
Comment (pditommaso) [inline modules/nextflow/src/main/groovy/nextflow/executor/TgccExecutor.groovy:2]: Technically you/your organization is the copyright holder

**Comment by lindenb** [comment]
> Nice contribution, however it would be nice to have some unit testing. You can check [SlurmExecutorTest](https://github.com/nextflow-io/nextflow/blob/19a72c402495e72df0140c6af05d299330fa5ba9/modules/nextflow/src/test/groovy/nextflow/executor/SlurmExecutorTest.groovy#L31-L31) as a reference

@pditommaso thanks, I'll check this later.

File: modules/nextflow/src/main/groovy/nextflow/executor/SlurmExecutor.groovy [hunk 2]
```diff
@@ -100,6 +100,11 @@ class SlurmExecutor extends AbstractGridExecutor {
 
     String getHeaderToken() { '#SBATCH' }
 
+    /** name of the submit command, overriden in TgccExecutor */
+    protected String getSubmidCmd() {
+    	return 'sbatch'
+    	}
```
Comment (bentsherman) [inline modules/nextflow/src/main/groovy/nextflow/executor/SlurmExecutor.groovy:106]:
```suggestion
    protected String getSubmitCommand() {
        return 'sbatch'
    }
```

**Review by bentsherman — CHANGES_REQUESTED** [review]

**Comment by bentsherman** [comment]
Check the indentation in your new class as well, it should be 4 spaces but I think there are tabs mixed in.

**Comment by lindenb** [comment]
I'm closing this. Looking at the source, I just realized that the [Bridge]( https://www.nextflow.io/docs/latest/executor.html#bridge ) Executor is  THE workflow manager used at the TGCC center :-)

**Comment by pditommaso** [comment]
Too many french executors!! 😆

**Pull request closed without merging**